scripts/microbe_stage/microbe_editor/microbe_editor.as

```diff
@@ -887,6 +896,33 @@ class MicrobeEditor{
         return (playerSpecies.generation+1);
     }
 
+    void getNucleus()
+    {
+        double lengthMicrobe = double(editedMicrobe.length());
+        for(uint i = 0; i < editedMicrobe.length(); ++i){
+            auto organelle = cast<PlacedOrganelle>(editedMicrobe[i]);
+            auto name = organelle.organelle.name;
+            if (name=="nucleus"){
+                nucleusIsPresent = 1;
+                LOG_INFO("nucleus is present? " + nucleusIsPresent);
+            }
+        }
+    }
+
+
+    void resetNucleus()
```

Comment:
Does this function work properly? How does it know that the nucleus was removed?

Reply:
That function just put the check value for nucleus to default if find a nucleus, is put in undo so when you are making undo after put nucleus you can put it again.
It's called every undo, so in every redu i check if nucleus is still present so i avoid putting more nucleus.

Reply:
But what I think can now happen is (if we tweak the mutation point values, the editor needs to be as general as possible to allow balancing) if you place a nucleus and then you have mp left to place an organelle, then if you undo that organelle this method notices that the cell still has a nucleus and marks that there is no nucleus present. Isn't this wrong logic?

There should be just one method for checking in the place organelle method that "does the cell currently have a nucleus".